src/renderer/mod.rs

```diff
@@ -238,18 +238,20 @@ impl Renderer {
         self.cursor_renderer
             .update_cursor_destination(font_dimensions.into(), windows);
 
-        self.cursor_renderer
+        animating |= self
+            .cursor_renderer
             .animate(&self.current_mode, &self.grid_renderer, dt);
 
         animating
     }
 
-    pub fn handle_draw_commands(&mut self) -> bool {
+    pub fn handle_draw_commands(&mut self) -> (bool, bool) {
```

Comment:
Nit: Consider returning a struct with named fields. Returning unlabeled booleans makes code hard to understand at a glance